File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperProforma.java
package com.friendlypos.principal.helpers;

import android.app.Activity;
import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver;
import com.friendlypos.application.datamanager.BaseManager;
import com.friendlypos.application.interfaces.RequestInterface;
import com.friendlypos.distribucion.modelo.EnviarFactura;
import com.friendlypos.distribucion.modelo.invoice;
import com.friendlypos.login.util.SessionPrefes;
import com.friendlypos.principal.activity.MenuPrincipal;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static android.content.ContentValues.TAG;

/**
 * Created by Delvo on 02/12/2017.
 */

public class SubirHelperProforma {

    private NetworkStateChangeReceiver networkStateChangeReceiver;
    private MenuPrincipal activity;
    private Context mContext;
    private RequestInterface mAPIService;
    String codigoS;
    String mensajeS;
    String resultS;
    int codigoServer;
    int codigo;
    public SubirHelperProforma(MenuPrincipal activity) {
        this.activity = activity;
        this.mContext = activity;
        networkStateChangeReceiver = new NetworkStateChangeReceiver();
        mAPIService = BaseManager.getApi();
    }

    public void sendPostProforma(EnviarFactura facturaQuery) {
        String token = "Bearer " + SessionPrefes.get(mContext).getToken();
        Log.d("tokenCliente", token + " ");
        if (isOnline()) {
            Log.d("factura1", facturaQuery + " ");
        mAPIService.savePostProforma(facturaQuery, token).enqueue(new Callback<invoice>() {



            @Override
            public void onResponse(Call<invoice> call, Response<invoice> response) {

                if(response.isSuccessful()) {
                    Log.d("respProforma",response.body().toString());
                    codigo = response.code();
                    codigoS = response.body().getCode();
                    mensajeS = response.body().getMessage();
                    resultS= String.valueOf(response.body().isResult());
                    activity.codigoDeRespuestaProforma(codigoS, mensajeS, resultS, codigo);

                }
                else{
                }
            }

            @Override
            public void onFailure(Call<invoice> call, Throwable t) {
                Log.e(TAG, "Unable to submit post to API.");
            }
        });}
        else{
            Toast.makeText(activity, "Error, por favor revisar conexión de Internet", Toast.LENGTH_SHORT).show();
        }
    }

    private boolean isOnline() {
        return networkStateChangeReceiver.isNetworkAvailable(mContext);
    }
}

File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperProforma.kt
package com.friendlysystemgroup.friendlypos.principal.helpers

import android.content.ContentValues
import android.content.Context
import android.util.Log
import android.widget.Toast
import com.friendlysystemgroup.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver
import com.friendlysystemgroup.friendlypos.application.datamanager.BaseManager.Companion.api
import com.friendlysystemgroup.friendlypos.application.interfaces.RequestInterface
import com.friendlysystemgroup.friendlypos.distribucion.modelo.EnviarFactura
import com.friendlysystemgroup.friendlypos.distribucion.modelo.invoice
import com.friendlysystemgroup.friendlypos.distribucion.modelo.invoice.isResult
import com.friendlysystemgroup.friendlypos.distribucion.modelo.invoice.message
import com.friendlysystemgroup.friendlypos.distribucion.modelo.invoice.toString
import com.friendlysystemgroup.friendlypos.login.util.SessionPrefes.Companion.get
import com.friendlysystemgroup.friendlypos.principal.activity.MenuPrincipal

/**
 * Created by Delvo on 02/12/2017.
 */
class SubirHelperProforma(private val activity: MenuPrincipal) {
    private val networkStateChangeReceiver =
        NetworkStateChangeReceiver()

    private val mContext: Context = activity
    private val mAPIService: RequestInterface? = api
    var codigoS: String? = null
    var mensajeS: String? = null
    var resultS: String? = null
    var codigoServer: Int = 0
    var codigo: Int = 0

    fun sendPostProforma(facturaQuery: EnviarFactura, cantidadFactura: String?) {
        val token = "Bearer " + get(mContext).token
        Log.d("tokenCliente", "$token ")
        if (isOnline) {
            Log.d("factura1", "$facturaQuery ")
            mAPIService!!.savePostProforma(facturaQuery, token)
                .enqueue(object : retrofit2.Callback<invoice?> {
                    override fun onResponse(
                        call: retrofit2.Call<invoice?>?,
                        response: retrofit2.Response<invoice?>
                    ) {
                        if (response.isSuccessful()) {
                            Log.d("respProforma", response.body().toString())
                            codigo = response.code()
                            codigoS = response.body().code
                            mensajeS = response.body().message
                            resultS = response.body().isResult.toString()
                            activity.codigoDeRespuestaProforma(
                                codigoS!!, mensajeS,
                                resultS!!, codigo, cantidadFactura
                            )
                        } else {
                        }
                    }

                    override fun onFailure(call: retrofit2.Call<invoice?>?, t: Throwable?) {
                        Log.e(ContentValues.TAG, "Unable to submit post to API.")
                    }
                })
        } else {
            Toast.makeText(
                activity,
                "Error, por favor revisar conexión de Internet",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private val isOnline: Boolean
        get() = networkStateChangeReceiver.isNetworkAvailable(mContext)
}
